core/src/main/java/com/linecorp/armeria/internal/common/TimeoutScheduler.java

```diff
@@ -39,6 +41,10 @@
     private EventLoop eventLoop;
     @Nullable
     private TimeoutController timeoutController;
+    @Nullable
+    private CompletableFuture<Void> timingOutFuture;
+    @Nullable
+    private CompletableFuture<Void> timedOutFuture;
```

Comment:
`volatile` and CAS please

Reply:
As commented in https://github.com/line/armeria/pull/2939#discussion_r460888125,
`sychronized` block is used.